### PR TITLE
refactor: split notifications to a new cdc handler

### PR DESCRIPTION
**Pull request by idoshamun**
To overcome the ordering constraint of CDC, we separate the notifications to a new topic to allow much faster handling with spikes

WT-2048 #done

### CODE REVIEW

**Review by rebelchris — APPROVED** [review]
Perhaps wait for a second set of eyes.
Looks ok, but not 100% sure on the debezium config myself.

**Comment by idoshamun** [comment]
@capJavert can you have another look please? you also fought debezium quite a lot

**Review by capJavert — APPROVED** [review]
Looks alright, one question, if we deploy this and for example have X notifications to process they would be lost due to original cdc no longer handling it or would Debezium re-route?

File: .infra/application.properties [hunk 1]
```diff
@@ -13,10 +13,13 @@ debezium.source.plugin.name=pgoutput
 debezium.source.heartbeat.interval.ms=60000
 debezium.source.topic.prefix=t
 debezium.source.tombstones.on.delete=false
-debezium.transforms=Reroute,ReadOperationFilter,PostsFilter
+debezium.transforms=Reroute,Notifications,ReadOperationFilter,PostsFilter
 debezium.transforms.Reroute.type=io.debezium.transforms.ByLogicalTableRouter
-debezium.transforms.Reroute.topic.regex=(.*)
+debezium.transforms.Reroute.topic.regex=^((?!\.notification_v2).)*$
```
Comment (capJavert) [inline .infra/application.properties:18]: Additionally, would this match, our topic does not have `_v2` in its name?

Comment (capJavert) [reply .infra/application.properties:18]: Or is this a table name match?

Comment (idoshamun) [reply .infra/application.properties:18]: This table name match. It took me a while to find the right regex but I tested everything locally. It was a tough fight. We no longer have v1 tables btw

Comment (capJavert) [reply .infra/application.properties:18]: roger, roger! good then 👍

**Review by capJavert — APPROVED** [review]

**Pull request merged**